bundles/org.eclipse.passage.lic.api/src/org/eclipse/passage/lic/api/internal/registry/Registry.java

```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ * Copyright (c) 2020 ArSysOp
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ArSysOp - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.passage.lic.api.internal.registry;
+
+import java.util.Collection;
+
+/**
+ * 
+ * @param <S> sub type of {@linkplain Service}
+ * @since 0.7
+ */
+public interface Registry<S extends Service<?>> {
+
+	boolean hasService(ServiceId id);
+
+	S service(ServiceId id);
+
+	Collection<S> services();
```

Comment:
List? I expect it ordered but cannot explain why.

Reply:
In general case order will be unpredictable eitherway. 

Indeed, default implementation of, say, `RuntimeRegistry` allows to install a particular implementation of physical storage (say, `TreeMap`), which potentially can be implicitely used for service ordering. 

But on the current depth of api design detalisation _ordering_  is not the feature of registries. It should 
 - either become one and beget more complex interfaces - but benefits are not clear
 - or copletely kept for clients - and in this case it looks better to precisely declare "do not rely on any order".

Does this convince you?

Reply:
ok